### PR TITLE
feat(plugin-lighthouse): add opportunities to lighthouse details

### PR DESCRIPTION
**Pull request by BioPhoton**
This PR add the `opportunity` detail to CP reports.

It includes:
- parsing logic for `opportunity` details
- error handling refactoring
- additional tests for uncovered parts of the lighthouse plugin

### CODE REVIEW

File: packages/plugin-lighthouse/src/lib/runner/constants.ts
Comment (matejchalk) [inline packages/plugin-lighthouse/src/lib/runner/constants.ts:1]: Please revert these changes. The previous version was safer - potential `null` value of `defaultConfig.audits` and `defaultConfig.categories` was safely handled by nullish coalescing (`audits ?? []` and `categories ?? []`). New version assumes `null` can't happen (on what basis?) and overrides Typescript with type-cast (`audits as Config.AuditJson[]` and `categories as Record<string, Config.CategoryJson>`).

File: packages/plugin-lighthouse/src/lib/runner/utils.unit.test.ts [hunk 1]
```diff
@@ -126,14 +128,45 @@ describe('toAuditOutputs', () => {
...
+            id: 'cumulative-layout-shift',
+            details: {
+              type: 'table',
+              headings: [{ key: 'number' }],
+              items: [{ number: 42 }],
+            },
+            score: 0,
+            numericValue: 0,
+            displayValue: '0 ms',
+          } as unknown as Result,
```
Comment (matejchalk) [inline packages/plugin-lighthouse/src/lib/runner/utils.unit.test.ts:145]: There's always a better way than `as unknown as T` (part 1):

```suggestion
          {
            id: 'cumulative-layout-shift',
            details: {
              type: 'table',
              headings: [{ key: 'number' }] as Details.TableColumnHeading[],
              items: [{ number: 42 }],
            } as Details,
            score: 0,
            numericValue: 0,
            displayValue: '0 ms',
          } as Result,
```

File: packages/plugin-lighthouse/src/lib/runner/utils.unit.test.ts [hunk 2]
```diff
@@ -126,14 +128,45 @@ describe('toAuditOutputs', () => {
...
         type =>
           ({
-            details: { type },
-          } as Result),
+            id: 'cumulative-layout-shift',
+            details: { type, headings: [], items: [] },
+            score: 0,
+            numericValue: 0,
+            displayValue: '0 ms',
+          } as unknown as Result),
```
Comment (matejchalk) [inline packages/plugin-lighthouse/src/lib/runner/utils.unit.test.ts:169]: There's always a better way than `as unknown as T` (part 2):

```suggestion
            id: 'cumulative-layout-shift',
            details: { type, headings: [], items: [] } as Details,
            score: 0,
            numericValue: 0,
            displayValue: '0 ms',
          } as Result),
```

File: packages/plugin-lighthouse/src/lib/runner/utils.unit.test.ts [hunk 3]
```diff
@@ -145,13 +178,61 @@ describe('toAuditOutputs', () => {
...
```
Comment (matejchalk) [inline packages/plugin-lighthouse/src/lib/runner/utils.unit.test.ts:186]: There's always a better way than `as unknown as T` (part 3):

```suggestion
            id: 'cumulative-layout-shift',
            details: { type, headings: [], items: [] } as Details,
            score: 0,
            numericValue: 0,
            displayValue: '0 ms',
          } as Result),
```

File: packages/plugin-lighthouse/src/lib/runner/utils.unit.test.ts [hunk 4]
```diff
@@ -145,13 +178,61 @@ describe('toAuditOutputs', () => {
...
+            id: 'cumulative-layout-shift',
+            details: {
+              type: 'table',
+              headings: [],
+              items: [],
+            },
+            score: 0,
+            numericValue: 0,
+            displayValue: '0 ms',
+          } as unknown as Result,
```
Comment (matejchalk) [inline packages/plugin-lighthouse/src/lib/runner/utils.unit.test.ts:207]: There's always a better way than `as unknown as T` (part 4):

```suggestion
          {
            id: 'cumulative-layout-shift',
            details: {
              type: 'table',
              headings: [],
              items: [],
            } as Details,
            score: 0,
            numericValue: 0,
            displayValue: '0 ms',
          } as Result,
```

File: packages/plugin-lighthouse/src/lib/runner/constants.ts
Comment (BioPhoton) [reply packages/plugin-lighthouse/src/lib/runner/constants.ts:1]: reverted

**Pull request merged**